src/components/connect-request.js

```diff
@@ -50,39 +67,77 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
             return this._wrappedInstance;
         }
 
-        cancelPendingRequests() {
+        // returns [[ "keys", "to", "cancel"], [{ config objects to request }]]
+        _diffConfigs(prevConfigs, configs) {
+            const unpack = (c) => {
+                return getQueryKey(c.url, c.body);
+            };
+            const prevQueryKeys = prevConfigs.map(unpack);
+            const queryKeys = configs.map(unpack);
+            const i = intersection(prevQueryKeys, queryKeys);
+            const cancelKeys = difference(prevQueryKeys, i);
+            const requestKeys = difference(queryKeys, i);
+
+            const toRequest = prevConfigs.filter((c) => {
+                const k = getQueryKey(c.url, c.body);
+                return requestKeys.includes(k);
+            });
+
+            return [cancelKeys, toRequest];
+        }
+
+        cancelPendingRequests(toCancel) {
             const { dispatch } = this.context.store;
+            if (toCancel && Array.isArray(toCancel)) {
+                // Only cancel specific keys, supports multiple configs
+                toCancel.forEach((queryKey) => {
+                    if (this._pendingRequests.hasOwnProperty(queryKey)) {
+                        dispatch(cancelQuery(queryKey));
+                    }
+                });
+            } else {
+                for (const queryKey in this._pendingRequests) {
+                    if (this._pendingRequests.hasOwnProperty(queryKey)) {
+                        dispatch(cancelQuery(queryKey));
+                    }
+                }
+            }
+        }
+
+        requestAsync(propsOrConfigs, force = false, retry = false) {
```

Comment:
I think this can be simplified if we always pass in `configs` to this function and combine with the `_makeRequest` method. So `requestAsync` always requires `configs` instead of reading from `props`, and it does the checking internally for when `configs` is an array or not. Also would be nice if when it is an array that we do a truthiness check on each config like we do when it's not an array (i.e. `configs.filter(Boolean)`).